book/python/writing-your-own-partitioned-stateful-application.md

```diff
@@ -6,10 +6,11 @@ In this section, we will go over how to write a partitioned stateful application
 
 Partitioning is a key aspect of how work is distributed in Wallaroo. From the application's point of view, what is required are:
 
-* a list of partition keys
 * a partitioning function
 * partition-compatible states - the state should be defined in such a way that each partition can have its own distinct state instance
 
+A list of partition keys is optional. If included, your application will use these keys. If not included, the application will dynamically create keys as needed. There is a slightly performance penalty to dynamically create keys. 
```

Comment:
`will use these keys to set up the initial state partitions in the system. As we encounter new keys derived from messages using the user-defined partition function, we will dynamically create new state partitions. There is a slight performance penalty to dynamically creating state partitions in this way, so if you know all your keys ahead of time, it can be a good idea to provide them here.`